File: MomentsJava/app/src/main/java/com/example/momentsjava/model/ListItem.java
package com.example.momentsjava.model;

public class ListItem {
    private UserInfo userInfo;
    private MomentInfo momentInfo;

    public ListItem(UserInfo userInfo, MomentInfo momentInfo) {
        this.userInfo = userInfo;
        this.momentInfo = momentInfo;
    }

    public UserInfo getUserInfo() {
        return userInfo;
    }

    public void setUserInfo(UserInfo userInfo) {
        this.userInfo = userInfo;
    }

    public MomentInfo getMomentInfo() {
        return momentInfo;
    }

    public void setMomentInfo(MomentInfo momentInfo) {
        this.momentInfo = momentInfo;
    }
}

File: MomentsKotlin/src/main/java/com/example/moments/data/model/ListItem.kt
package com.example.moments.data.model

import androidx.annotation.Keep
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
@Keep
data class ListItem(
    @PrimaryKey(autoGenerate = true) val id: Int = 0,
    @Embedded val userInfo: UserInfo,
    @Embedded val momentInfo: MomentInfo
)

@Keep
data class UserInfo(
    val username: String,
    val avatar: String
)

@Keep
data class MomentInfo(
    val text: String,
    val picture: List<String>
)
